enumerable_project.rb

```diff
@@ -0,0 +1,139 @@
+ Copyright (c) Muhammasd Arslan
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each do |x|
+      arr << x if yield(x) == true
+    end
+    arr
+  end
+
+  def my_all?
+    return self unless block_given?
+
+    my_each do |x|
+      return false if yield(x) == false
+    end
+    true
+  end
+
+  def my_any?
+    return self unless block_given?
+
+    my_each do |x|
+      return true if yield(x) == true
+    end
+    false
+  end
+
+  def my_none?
+    return self unless block_given?
+
+    my_each do |x|
+      return false if yield(x) == true
+    end
+    true
+  end
+
+  def my_count(*args)
+    if args.empty?
+      return length unless block_given?
+
+      i = 0
+      my_each do |x|
+        i += 1 if yield(x) == true
+      end
+      i
+    else
+      i = 0
+      my_each do |x|
+        i += 1 if args[0] == x
+      end
+      i
+    end
+  end
+
+  def my_map
+    return self if proc.nil? && !block_given?
+
+    arr = []
+    my_each do |x|
+      val = proc.nil? ? yield(x) : proc.call(x)
+      arr << val
+    end
+    arr
+  end
+
+  def my_inject(acc = self[0])
+    return self unless block_given?
+
+    my_each do |x|
+      if x == acc
+        next
+      else
+        acc = yield(acc, x)
+      end
+    end
+    acc
+  end
+end
```

Comment:
my_inject:

When a symbol is specified combines each element of the collection by applying the symbol as a named method
`p (5..10).my_inject(2, :*)`  should return 302400

Combines all elements of enum by applying a binary operation, specified by a block:
`p (1..5).my_inject(4) { |prod, n| prod * n }` should return 480